app/main.py

```diff
@@ -1,2 +1,123 @@
+from typing import List, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, elements: List[tuple] = None) -> None:
+        if elements is None:
+            elements = []
+        self._bucket_size = 8
+        self._bucket_resize = self._bucket_size * 2 / 3
+        self.length = len(elements)
+        self._creating_buckets(self.length)
+        if self.length:
+            self._assign_buckets(elements)
+
+    def _creating_buckets(self, len_elements: int = 0) -> None:
+        while len_elements > self._bucket_resize:
+            self._bucket_size *= 2
+            self._bucket_resize = self._bucket_size * 2 / 3
+
+        self._buckets = [[] for i in range(self._bucket_size)]
+
+    def _assign_buckets(self, elements: List[tuple]) -> None:
+        for _key, _value in elements:
+            if isinstance(_key, list | dict | set):
+                raise TypeError(
+                    f'The key can`t be like this: "{_key}" is "{type(_key)}"'
+                )
+            hashed_value = hash(_key)
+            index = hashed_value % self._bucket_size
+
+            while self._buckets[index]:
+                if self._buckets[index][0] == _key:
+                    break
+                index = (index + 1) % self._bucket_size
+
+            self._buckets[index] = (_key, _value,)
+            self.length += 1
+
+    def _get_buckets_full(self) -> list[Any]:
+        return [
+            full_bucket for full_bucket in self._buckets
+            if full_bucket and full_bucket[1] != KeyError
```

Comment:
What is this `full_bucket[1] != KeyError`?

Reply:
It is for delitems

Reply:
It is wrong solution then
If something will be delated then it must be as start value